source/Halibut/Transport/Protocol/ResponseMessage.cs

```diff
@@ -27,7 +27,17 @@ public static ResponseMessage FromError(RequestMessage request, string message)
 
         public static ResponseMessage FromException(RequestMessage request, Exception ex)
         {
-            return new ResponseMessage {Id = request.Id, Error = new ServerError { Message = ex.UnpackFromContainers().Message, Details = ex.ToString() }};
+            return new ResponseMessage {Id = request.Id, Error = ServerErrorFromException(ex)};
+        }
+
+        internal static ServerError ServerErrorFromException(Exception ex)
```

Comment:
Are we missing a test to guarantee that we are not breaking the current behaviour of this method, e.g. Old Halibut Client is compatible with new Halibut Service?

Reply:
Is that the case that a new Halibut service throws an exception and so ends up here and so adds the ErrorType to the ServerError. Then Old halibut receives a ServerError which contains an additional parameter.

What would that test be? I guess the test that might be possible to write is one that verifies the message serializer can successfully ignore that extra parameter. I will add this test.

The other test would be an integration test between new Halibut and old Halibut. I am still unsure how to piratically do that.

Reply:
> The other test would be an integration test between new Halibut and old Halibut. I am still unsure how to piratically do that.

Yes this would be the nicest but we need to figure that out. Maybe skip for now but something we can come back to later

Reply:
Added a test which shows that extra fields in ServerError are ignored by the serializer. This should at least show that for older halibut clients they will just ignore the extra field and just throw a HalibutClientException as they normaly would.